contracts/cw4-stake/src/contract.rs

```diff
@@ -239,20 +262,32 @@ pub fn execute_claim(
     }
 
     let config = CONFIG.load(deps.storage)?;
-    let amount;
-    match &config.denom {
-        Denom::Native(denom) => amount = coins(release.u128(), denom),
-        Denom::Cw20(_canonical_addr) => {
-            unimplemented!("The CW20 coins release functionality is in progress")
+    let (amount_str, message) = match &config.denom {
+        Denom::Native(denom) => {
+            let amount = coins(release.u128(), denom);
+            let amount_str = coins_to_string(&amount);
+            let message = BankMsg::Send {
+                to_address: info.sender.clone().into(),
+                amount,
+            }
+            .into();
+            (amount_str, message)
         }
-    }
-
-    let amount_str = coins_to_string(&amount);
-    let messages = vec![BankMsg::Send {
-        to_address: info.sender.clone().into(),
-        amount,
-    }
-    .into()];
+        Denom::Cw20(canonical_addr) => {
+            let amount_str = format!("{}{}", release.u128(), canonical_addr);
+            let transfer = Cw20ExecuteMsg::Transfer {
+                recipient: info.sender.clone().into(),
+                amount: release,
+            };
+            let message = WasmMsg::Execute {
+                contract_addr: canonical_addr.into(),
```

Comment:
```suggestion
       Denom::Cw20(addr) => {
            let amount = coins(release.u128(), addr);
            let amount_str = coins_to_string(&amount);
            let transfer = Cw20ExecuteMsg::Transfer {
                recipient: info.sender.to_string(),
                amount: release,
            };
            let message = WasmMsg::Execute {
                contract_addr: addr.into(),
```

Reply:
Also, in `coins_to_string()` below, the format might better be:
```rust
.map(|c| format!("{} {}", c.amount, c.denom))
```
with a space between amount and denom; just for clarity.

Reply:
Changed the format and used the same function to format in both variants of the match